torch/jit/_trace.py

```diff
@@ -976,6 +976,8 @@ def is_tracing():
     Returns ``True`` in tracing (if a function is called during the tracing of
     code with ``torch.jit.trace``) and ``False`` otherwise.
     """
+    if is_scripting():
```

Comment:
Is that expected to always be true? We can never trace in script mode?

Reply:
I'm not so sure. There is a "script in trace mode", but I'm not sure if it is or will be possible to do "trace in script mode" because trace requires extra inputs.

Reply:
We should make sure that this is the expected behavior?
cc @suo

Reply:
They should be mutually exclusive. Even though you can mix tracing and scripting (e.g. by tracing a module and then scripting its parent module or vice-versa) you can't do both at the same time.